server/chat-plugins/datasearch.ts

```diff
@@ -10,15 +10,59 @@
 
 'use strict';
 
+import {QueryProcessManager} from '../../lib/process-manager';
+import {Repl} from '../../lib/repl';
+import {Chat} from '../chat';
+import * as CL from '../config-loader';
+import {Dex} from '../../sim/dex';
```

Comment:
Where were these imports previously? Isn't this file supposed to just use the global one?

Reply:
Child processes manually imported all of them

Reply:
I think the child processes should continue doing that, to allow for monkeypatching and hotpatching.

Reply:
You should `ts-ignore` or whatever as necessary for that.

Reply:
Tried doing this with tslint dsabling and ts-ignoring where necessary and got this error
```
CRASH: TypeError: Dex.includeData is not a function
    at Object.<anonymous> (C:\Users\feeds\Desktop\Pokemon-Showdown\.server-dist\chat-plugins\datasearch.js:2049:6)
    at Module._compile (internal/modules/cjs/loader.js:777:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:788:10)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:840:10)
    at internal/main/run_main_module.js:17:11
```

Reply:
You're importing the wrong thing as Dex. You're probably doing `require('..dex').includeData()` instead of `require('..dex').Dex.includeData()`

Reply:
yep, fixed